File: app/src/main/java/br/com/pedrosilva/tecnonutri/domain/interactors/GetFeedItemInteractor.java
package br.com.pedrosilva.tecnonutri.domain.interactors;

import br.com.pedrosilva.tecnonutri.domain.entities.FeedItem;

/**
 * Created by psilva on 3/20/17.
 */

public interface GetFeedItemInteractor {
    interface Callback {
        void onFeedItemRetrieved(FeedItem feedItem);
        void onFeedItemRetrieveFailed(Throwable t);
    }
}

File: app/src/main/kotlin/br/com/pedrosilva/tecnonutri/domain/interactors/GetFeedItemInteractor.kt
package br.com.pedrosilva.tecnonutri.domain.interactors

import br.com.pedrosilva.tecnonutri.domain.entities.FeedItem

interface GetFeedItemInteractor {
    interface Callback {
        fun onFeedItemRetrieved(feedItem: FeedItem)
        fun onFeedItemRetrieveFailed(error: Throwable)
    }
}